### DISCUSSION TITLE
allowedVersions, version ranges, and lock files?

### DISCUSSION THREAD

**Discussion by jdfm** [renovatebot/renovate#13476] [Request Help]
Hello all, I have a config that looks like this in my renovate.json config file:

```json
    {
      "description": "breaking change avoidance",
      "matchManagers": ["npm"],
      "matchPackagePatterns": [
        "packagenamexyz"
      ],
      "allowedVersions": "< 2.0.4 || >= 3"
    },
```

In the PR/MR description it lists the version I'm expecting it to install `2.0.2 -> 2.0.3` as what's going to be upgraded to, but, in the changelist I see in the `yarn.lock` file that it's actually going to install `2.1.2` instead. It's also interesting to note that the changes are only happening in the `yarn.lock` file for this collection of updates, which hasn't been the case up to now.

I think part of the issue is that this specific package is part of "monorepo" config I had set up for some internal and private libraries, and I only noticed recently that they weren't pinned (where others in the package.json were). And the version specified in the `package.json` for this specific package is `^2.0.2`.

I guess my expectation was that, if there is a version specified like `^2.0.2` and a config like `"allowedVersions": "< 2.0.4 || >= 3"` that it would choose `2.0.3` as the largest version allowed by the specified version in the `package.json` that still passes the `allowedVersions` criteria.

**Comment by jdfm** [comment]
The version in the lock file, before the changes suggested by renovate, is `2.0.2` for this package, so, it's not because the version specified there is already larger.

I'm thinking maybe I need to add some extra configs to the monorepo config I set up, but, I'm not sure.

**Reply by rarkins** [reply]
To answer this, debug logs would be needed from when the commit is made, and more information on your config.

**Comment by viceice** [comment]
I think yarn is updating the lockfile to that version, as it's allowed by package.json.

We run `yarn install` after any update.

**Comment by jdfm** [comment]
The issue may be how I configured the monorepo stuff, specifically around the `rangeStrategy`...

```json
    {
      "description": "group XYZ package updates",
      "matchManagers": ["npm"],
      "matchPackagePrefixes": ["XYZ"],
      "groupName": "XYZ monorepo",
      "rangeStrategy": "update-lockfile"
    },
```

I think I did this originally because it looked like renovate at the time wasn't identifying the versions that were installed in the lockfile for these packages and this seemed to fix it for me. But, I realize now, this is probably interfering with how dependency pinning works and now these `allowedVersions` as well.
